include/xtensor-io/xfile_array.hpp

```diff
@@ -609,10 +631,10 @@ namespace xt
     template <class E, class IOH>
     inline void xfile_array_container<E, IOH>::flush()
     {
-        if (m_dirty)
+        if (m_dirty != xfile_dirty(false))
```

Comment:
I think a bool operator conversion in the `xfile_dirty` struct would allow the more idiomatic syntax

```cpp
if (m_dirty)
{
    .....
}
```

Reply:
Indeed, thanks.